src/__tests__/a11yAPI.test.js

```diff
@@ -62,133 +62,255 @@ function Section() {
+  const buttonAlias = queryByHintText(/button/g);
+  expect(buttonAlias && buttonAlias.props.accessibilityHint).toEqual(
+    BUTTON_HINT
+  );
 
   expect(() => getByA11yHint(NO_MATCHES_TEXT)).toThrow(NO_INSTANCES_FOUND);
+  expect(() => getByHintText(NO_MATCHES_TEXT)).toThrow(NO_INSTANCES_FOUND);
   expect(queryByA11yHint(NO_MATCHES_TEXT)).toBeNull();
+  expect(queryByHintText(NO_MATCHES_TEXT)).toBeNull();
 
   expect(() => getByA11yHint(TEXT_HINT)).toThrow(FOUND_TWO_INSTANCES);
+  expect(() => getByHintText(TEXT_HINT)).toThrow(FOUND_TWO_INSTANCES);
   expect(() => queryByA11yHint(TEXT_HINT)).toThrow(FOUND_TWO_INSTANCES);
+  expect(() => queryByHintText(TEXT_HINT)).toThrow(FOUND_TWO_INSTANCES);
 
   const asyncButton = await findByA11yHint(BUTTON_HINT);
   expect(asyncButton.props.accessibilityHint).toEqual(BUTTON_HINT);
+  const asyncButtonAlias = await findByA11yHint(BUTTON_HINT);
+  expect(asyncButtonAlias.props.accessibilityHint).toEqual(BUTTON_HINT);
   await expect(findByA11yHint(NO_MATCHES_TEXT, waitForOptions)).rejects.toThrow(
     NO_INSTANCES_FOUND
   );
+  await expect(findByHintText(NO_MATCHES_TEXT, waitForOptions)).rejects.toThrow(
+    NO_INSTANCES_FOUND
+  );
   await expect(findByA11yHint(TEXT_HINT, waitForOptions)).rejects.toThrow(
     FOUND_TWO_INSTANCES
   );
+  await expect(findByHintText(TEXT_HINT, waitForOptions)).rejects.toThrow(
+    FOUND_TWO_INSTANCES
+  );
 });
 
-test('getAllByA11yHint, queryAllByA11yHint', async () => {
-  const { getAllByA11yHint, queryAllByA11yHint, findAllByA11yHint } = render(
-    <Section />
-  );
+test('getAllByA11yHint, getAllByHintText, queryAllByA11yHint, queryAllByHintText, findAllByA11yHint, findAllByHintText', async () => {
+  const {
+    getAllByA11yHint,
+    getAllByHintText,
+    queryAllByA11yHint,
+    queryAllByHintText,
+    findAllByA11yHint,
+    findAllByHintText,
+  } = render(<Section />);
 
   expect(getAllByA11yHint(TEXT_HINT)).toHaveLength(2);
+  expect(getAllByHintText(TEXT_HINT)).toHaveLength(2);
   expect(queryAllByA11yHint(/static/g)).toHaveLength(2);
+  expect(queryAllByHintText(/static/g)).toHaveLength(2);
 
   expect(() => getAllByA11yHint(NO_MATCHES_TEXT)).toThrow(NO_INSTANCES_FOUND);
+  expect(() => getAllByHintText(NO_MATCHES_TEXT)).toThrow(NO_INSTANCES_FOUND);
   expect(queryAllByA11yHint(NO_MATCHES_TEXT)).toEqual([]);
+  expect(queryAllByHintText(NO_MATCHES_TEXT)).toEqual([]);
 
   await expect(findAllByA11yHint(TEXT_HINT)).resolves.toHaveLength(2);
+  await expect(findAllByHintText(TEXT_HINT)).resolves.toHaveLength(2);
   await expect(findAllByA11yHint(NO_MATCHES_TEXT)).rejects.toThrow(
     NO_INSTANCES_FOUND
   );
+  /**
+   * FIXME:
+   * Additional await expect yields the following:
+   * Expected substring: "No instances found"
+   * Received message:   "Unable to find node on an unmounted component."
+   */
+  // await expect(findAllByHintText(NO_MATCHES_TEXT)).rejects.toThrow(
+  //   NO_INSTANCES_FOUND
+  // );
```

Comment:
How about not touching the existing tests and only checking if `findByA11yHint` et. al. reference the same function as `finByHintText`? These are aliases, so we only if it's possible to use them (with typings tests) and they do exactly the same thing (by checking if they're referentially equal).

Reply:
This would likely be a lot cleaner than the current approach. Plus, it gives us the confidence that the aliases are referring to the same thing. I'll push an update for this soon, thanks again!